CMakeLists.txt

```diff
@@ -15,6 +15,8 @@ project(opentelemetry-cpp)
 # Mark variables as used so cmake doesn't complain about them
 mark_as_advanced(CMAKE_TOOLCHAIN_FILE)
 
+set(CMAKE_EXPORT_COMPILE_COMMANDS ON)
```

Comment:
Should this be enabled only for CMake v3.5 or greater?

Reply:
Valid point — just added a version check since the CMake variable wasn't introduced until 3.5.